src/components/numberinput/Numberinput.vue

```diff
@@ -153,7 +153,9 @@ export default {
                     }
                 }
                 this.newValue = newValue
-                this.$emit('input', newValue)
+                if (!isNaN(newValue) && newValue !== null && newValue !== '-0') {
```

Comment:
Purpose is only emit when `newValue` can be cast to a valid number, also `-0` will be cast to `0` without negative sign which will cause input becomes 0, so this case is ignored here.